### ISSUE TITLE
Performance issue: parseCss

### ISSUE THREAD

**Issue by Karolusrex** [d3/d3-interpolate#57]
The parseCss method transforms a statement like `translate(23px, 24px)` into a transformation matrix. It constantly modifies the DOM. This is a huge performance issue, when an animation is initiated. 

https://github.com/d3/d3-interpolate/blob/master/src/transform/parse.js

What do you think about parsing this with regex instead and mathematically generating the transformation matrices? Should be significantly faster. We can look for inspiration in places like https://github.com/jlmakes/rematrix for matrix generation. An simple example where performance is already a huge drain:

```
<html>
<head>
    <title>Learn D3 in 5 minutes</title>
</head>
<body>


<svg id="canvas"></svg>
<script src="https://d3js.org/d3.v5.js"></script>
<script>
    const circles = d3.select('svg')
        .selectAll('path')
        .data(Array(2).fill())
        .enter()
        .append('circle')
        .attr('r', 20)
    
    let xOffset = 0;
    const animate = () => {
        xOffset+=0.1
        circles.transition()
        .duration(100)
        .style('transform', (_, index) => `translate(${(index + xOffset) * 30}px, 30px)`)
        .on('end', animate);
    }
    animate();
</script>
</body>
</html>
```

On my brand new Macbook Pro I can see the framedrops easily with my eyes. As seen in Chrome, this triggers forced reflows. For a lot of simultaneous transitions starting at the same time, the problem is, of course, even bigger. 

<img width="287" alt="pastedgraphic-3" src="https://user-images.githubusercontent.com/2648614/41024909-c4cb1000-6970-11e8-83e1-a11ab49e9e00.png">

Is this a known issue? I would be interested in helping out with a pull request

**Comment by Karolusrex** [comment]
Turns out that there is a native API for doing this, working at least in latest Chrome. The window.DOMMatrix is a class for parsing these strings: https://developer.mozilla.org/en-US/docs/Web/API/DOMMatrix

**Comment by mbostock** [comment]
Sorry for the slow response.

Unfortunately I’m not sure that DOMMatrix buys us anything: what interpolateTransform *should* be doing is [interpolating a transform list](https://www.w3.org/TR/css-transforms-1/#interpolation-of-transforms) rather than a single transform matrix; see #44. And while SVG provides a native [SVGTransformList interface](https://developer.mozilla.org/en-US/docs/Web/API/SVGTransformList) I don’t see that there’s a planned replacement for that will cover CSS as well.

Especially vexing is that getComputedStyle always returns the matrix(…) form for transforms, which means that doesn’t appear to be a way to get the current style as a transform list, and that means that it’s probably impossible to implement transform list interpolation for d3-transition, even if it’s supported in d3-interpolate.

**Comment by mbostock** [comment]
Though… if we abandon our goal of interpolating transform lists and instead continue to interpolate transform matrices, then adopting DOMMatrix (or WebKitCSSMatrix) to parse seems like a win.

**Comment by mbostock** [comment]
Untested branch here:

https://github.com/d3/d3-interpolate/tree/DOMMatrix

**Comment by Karolusrex** [comment]
Yes, exactly like that! That is very similar to the code I'm using as well. 

Like this: https://github.com/Karolusrex/d3-interpolate/commit/edfdaa6147b03d4f7a3c6e18abe8b4104f016233

I understand the value in interpolating values instead of interpolating the matrices they represent. Perhaps interpolating the transition as matrix could be left as an option, once the feature to interpolate the values inbetween as landed. In both cases, there is no need to un-performance it by modifying the DOM.
